treenode.go

```diff
@@ -368,10 +378,16 @@ func (n *TreeNodeInstance) DispatchChannel(msgSlice []*ProtocolMsg) error {
 		reflect.ValueOf(n.channels[mt]).Send(out)
 	} else {
 		for _, msg := range msgSlice {
-			out := n.channels[mt]
+			out := reflect.ValueOf(n.channels[mt])
 			m := n.reflectCreate(to.Elem(), msg)
 			log.Lvl4(n.Name(), "Dispatching msg type", mt, " to", to, " :", m.Field(1).Interface())
-			reflect.ValueOf(out).Send(m)
+			if out.Len() < out.Cap() {
+				out.Send(m)
+			} else {
+				return fmt.Errorf("channel too small for msg %s in %s: "+
```

Comment:
I would even panic in this case. What do you think ?

Reply:
that would mean that a rogue node could panic other nodes... I prefer not to panic there. Ideally it should stop the protocol, but I did not implement that.